### PR TITLE
Improve test coverage

### PR DESCRIPTION
**Pull request by lquerel**
Test coverage improvements (#40):
- More tests for the metrics (optional description, unit, sum, min, max)
- Description of the testing and validation plan (see README.md)

In addition to test coverage, this PR also updated
- the metrics benchmark CLI tool,
- the arrow_record package to make the new metrics encoder/decoder accessible for the experimental collector.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,7 +16,34 @@ Other important links:
...
+
+A validation of the compression ratio stability is also part of the objectives. This validation will be performed on production data.
+
+## Security
+
+A thread model is being defined [WIP] (untrusted input data, what can go wrong at the protocol level, during the 
+encoding or decoding phases, ...). Below the main risks identified so far:
+- invalid, or compromised inputs causing security or reliability issues.
+- very large input data causing denial of service.
+- high cardinality data causing dictionary overflow (over multiple messages).
```
Comment (jmacd) [inline README.md:42]: fwiw, there's an active discussion which I am trying to resolve here: https://github.com/open-telemetry/opentelemetry-specification/issues/1891

File: pkg/datagen/data_generator.go [hunk 2]
```diff
@@ -106,6 +137,60 @@ func (dg *DataGenerator) GenI64Range(min int64, max int64) int64 {
...
+}
+
+func (dg *DataGenerator) HasHistogramSum() bool {
+	if dg.config.ProbHistogramHasSum == 1.0 {
+		return true
+	} else if dg.config.ProbHistogramHasSum == 0.0 {
+		return false
+	} else {
+		return dg.rng.Float64() < dg.config.ProbHistogramHasSum
+	}
```
Comment (jmacd) [inline pkg/datagen/data_generator.go:167]:
```suggestion
	return dg.rng.Float64() < dg.config.ProbHistogramHasSum
```

File: pkg/datagen/data_generator.go [hunk 3]
```diff
@@ -106,6 +137,60 @@ func (dg *DataGenerator) GenI64Range(min int64, max int64) int64 {
...
+}
+
+func (dg *DataGenerator) HasHistogramMin() bool {
+	if dg.config.ProbHistogramHasMin == 1.0 {
+		return true
+	} else if dg.config.ProbHistogramHasMin == 0.0 {
+		return false
+	} else {
+		return dg.rng.Float64() < dg.config.ProbHistogramHasMin
+	}
```
Comment (jmacd) [inline pkg/datagen/data_generator.go:177]:
```suggestion
	return dg.rng.Float64() < dg.config.ProbHistogramHasMin
```

File: pkg/datagen/data_generator.go [hunk 4]
```diff
@@ -106,6 +137,60 @@ func (dg *DataGenerator) GenI64Range(min int64, max int64) int64 {
...
+}
+
+func (dg *DataGenerator) HasHistogramMax() bool {
+	if dg.config.ProbHistogramHasMax == 1.0 {
+		return true
+	} else if dg.config.ProbHistogramHasMax == 0.0 {
+		return false
+	} else {
+		return dg.rng.Float64() < dg.config.ProbHistogramHasMax
+	}
```
Comment (jmacd) [inline pkg/datagen/data_generator.go:187]:
```suggestion
	return dg.rng.Float64() < dg.config.ProbHistogramHasMax
```

**Review by jmacd — APPROVED** [review]
Just a few suggested expression simplifications.

File: README.md (hunk 1 above)
Comment (lquerel) [reply README.md:42]: Added a link to this issue in this document for follow-up purposes.

**Pull request merged**